gadget/install/install.go

```diff
@@ -452,6 +452,13 @@ func applyLayoutToOnDiskStructure(onDiskVol *gadget.OnDiskVolume, partNode strin
 	}
 	logger.Noticef("laidOutStruct.OnDiskStructure: %+v, *onDiskStruct: %+v",
 		laidOutStruct.OnDiskStructure, *onDiskStruct)
+
+	// Keep wanted filesystem label and type, as that is what we actually want
+	// on the disk.
+	if creatingPart {
+		onDiskStruct.PartitionFSType = laidOutStruct.PartitionFSType
+		onDiskStruct.PartitionFSLabel = laidOutStruct.PartitionFSLabel
+	}
```

Comment:
are there tests that notice this? as I don't see test changes in this commit itself

Reply:
Yes, the `muinstaller-real:encrypted` spread test fails if we are not careful here. When encrypting the partition and getting to this function, the fs type/label that we have read from disk are not the final ones as we have not encrypted it yet.

Reply:
I've modified now `MockGadgetPartitionedDisk()` so it does not fill the filesystem label, that actually triggers the bug in the unit tests (fs label is not expected to be set at that point in time, at least in some of the unit tests in gadget/install).